### PR TITLE
New AkkaStream module

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

File: Migration26.md [hunk 1]
```diff
@@ -27,6 +27,34 @@ The `.getFilledStatement` has been removed.
 
 The former streaming operation `.apply()` is now removed, and must be replaced by either `.fold`, `.foldWhile` or `.withResult`.
 
+### Akka Stream module
+
+A new [Akka](http://doc.akka.io/docs/akka/2.4.2/scala/stream/index.html) module is available to process DB results as [Sources](doc.akka.io/api/akka/2.4.2/#akka.stream.javadsl.Source).
```
Comment (marcospereira) [inline Migration26.md]: Should link to akka docs 2.4.4 instead of 2.4.2.

Comment (cchantep) [inline Migration26.md]: Updated

**Comment by schmitch** [comment]
Actually I tested it and it runs through tables pretty good, however after running through the whole result set i get the following exception:

```
[ERROR] [05/16/2016 11:19:06.748] [default-akka.actor.default-dispatcher-4] [akka://default/user/StreamSupervisor-0/flow-0-0-unknown-operation] Error during postStop in [anorm.AkkaStream$ResultSource@7a1afca]: Dieses ResultSet ist geschlossen.
org.postgresql.util.PSQLException: Dieses ResultSet ist geschlossen.
        at org.postgresql.jdbc.PgResultSet.checkClosed(PgResultSet.java:2725)
        at org.postgresql.jdbc.PgResultSet.getStatement(PgResultSet.java:684)
        at anorm.AkkaStream$ResultSource$$anon$1.anorm$AkkaStream$ResultSource$$anon$$release(AkkaStream.scala:112)
        at anorm.AkkaStream$ResultSource$$anon$1.postStop(AkkaStream.scala:104)
        at akka.stream.impl.fusing.GraphInterpreter.finalizeStage(GraphInterpreter.scala:680)
        at akka.stream.impl.fusing.GraphInterpreter.afterStageHasRun(GraphInterpreter.scala:659)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:550)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:469)
        at akka.stream.impl.fusing.GraphInterpreterShell.receive(ActorGraphInterpreter.scala:415)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:593)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:584)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:605)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:482)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:519)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
        at akka.actor.ActorCell.invoke(ActorCell.scala:495)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

Also if the ResultSet Parser has a wrongly named Parsing Column you would get the following exception:
```[ERROR] [05/16/2016 11:20:57.339] [default-akka.actor.default-dispatcher-4] [akka://default/user/StreamSupervisor-0/flow-0-0-unknown-operation] Error during postStop in [anorm.AkkaStream$ResultSource@618654b7]: Dieses ResultSet ist geschlossen.```

It's german and means the ResultSet is closed, but the Exception says nothing about what really happend.

Also my code to test was pretty simple:
```
    Class.forName("org.postgresql.Driver")
    implicit val actorSystem = ActorSystem()
    implicit val mat = ActorMaterializer()
    implicit val connection = DriverManager.getConnection("jdbc:postgresql://localhost/loki?user=loki&password=loki")
    try {
      val parser = (SqlParser.long("id") ~ SqlParser.str("hash")).map {
        case id ~ hash => (id, hash)
      }
      val table = "history"
      val count = SQL"SELECT COUNT(*) FROM #$table".as(SqlParser.scalar[Long].single)

      val atomic = new AtomicLong(0L)
      AkkaStream.source(SQL"SELECT * FROM #$table", parser)
              .runForeach{ case (id, hash) =>
                val curr = atomic.incrementAndGet()
                println(s"ID: $id :: HASH: $hash :: CURR: $curr")
              }

      while(atomic.get() != count) {
      }
    } finally {
      connection.close()
      actorSystem.terminate()
    }
```

**Comment by cchantep** [comment]
Will check. As the next row is requested on demand/pull, for me the back pressure should be ok.

**Comment by cchantep** [comment]
@schmitch you can give a try to the update PR pls

**Comment by schmitch** [comment]
@cchantep now it works fine.
LGTM

**Comment by cchantep** [comment]
Thanks @schmitch for the review

**Comment by cchantep** [comment]
Thanks @gmethvin

**Comment by schmitch** [comment]
Could we make a release soon, I would really need the library.

**Comment by cchantep** [comment]
I guess it's better to ask on gitter (I'm not part of the release process).

**Comment by gmethvin** [comment]
@cchantep I can do releases, but I don't monitor this project that closely so you're probably a better judge of when to do it. You can just open an issue like you did for 2.5.1 and @mention me.

**Pull request merged**